openpype/hosts/resolve/api/plugin.py

```diff
@@ -405,26 +405,41 @@ def load(self, files):
             self.active_bin
         )
         _clip_property = media_pool_item.GetClipProperty
+        source_in = int(_clip_property("Start"))
```

Comment:
How about this?

```python

        if not self.with_handles:
            # Load file without the handles of the source media
            # We remove the handles from the source in and source out
            # so that the handles are excluded in the timeline
            handle_start = 0
            handle_end = 0

            # get version data frame data from db
            version_data = self.data["versionData"]
            frame_start = version_data.get("frameStart")
            frame_end = version_data.get("frameEnd")

            # The version data usually stored the frame range + handles of the 
            # media however certain representations may be shorter because they
            # exclude those handles intentionally. Unfortunately the 
            # representation does not store that in the database currently; 
            # so we should compensate for those cases. If the media is shorter 
            # than the frame range specified in the database we assume it is 
            # without handles and thus we do not need to remove the handles 
            # from source and out
            if frame_start is not None and frame_end is not None:
                # Version has frame range data, so we can compare media length
                handle_start = version_data.get("handleStart", 0)
                handle_end = version_data.get("handleEnd", 0)
                frame_start_handle = frame_start - handle_start
                frame_end_handle = frame_start + handle_end
                database_frame_duration = int(
                    frame_end_handle - frame_start_handle + 1
                )
                if source_duration >= database_frame_duration:
                    source_in += handle_start
                    source_out -= handle_end

        # get timeline in
        timeline_start = self.active_timeline.GetStartFrame()
        if self.sequential_load:
            # set timeline start frame
            timeline_in = int(timeline_start)
        else:
            # set timeline start frame + original clip in frame
            timeline_in = int(
                timeline_start + self.data["assetData"]["clipIn"])

        # make track item from source in bin as item
        timeline_item = lib.create_timeline_item(
            media_pool_item,
            self.active_timeline,
            timeline_in,
            source_in,
            source_out,
        )
```

Reply:
Yeah that might be good to do similarly. How about this? 2c0d1b5c2bd3efb9e0740b0d5324a588533459e1

Reply:
I personally prefer the `if not self.with_handles` nesting I did just to make it very clear that _all that_ is really only needed for that case - your logic still goes back/forth leaving me wonder a lot WHY we're doing a lot of the stuff.

But that might be just me.

Reply:
We don't need the handles at all if `self.with_handles` is not true.

As such that example should be:
```python


# check if user want handles 
if self.with_handles:
    version_data = self.data["versionData"]
    frame_start = version_data.get("frameStart")
    frame_end = version_data.get("frameEnd")
    handle_start = version_data.get("handleStart", 0)
    handle_end = version_data.get("handleEnd", 0)

    # check if representation has enough source for handles to be loaded
    frame_start_handle = frame_start - handle_start
    frame_end_handle = frame_start + handle_end
    database_frame_duration = int(
          frame_end_handle - frame_start_handle + 1
    )
    if source_duration < database_frame_duration:
         # once source duration is shorter, a representation is missing handles
         handle_start = 0
         handle_end = 0

    # trim source duration without handles
    source_in += handle_start
    source_out -= handle_end
```

Making it clear that source in and source out is only adjust ever if that if statement is True.